packages/ui/src/containers/VerificationCode/use-link-social-confirm-modal.ts

```diff
@@ -0,0 +1,46 @@
+import { SignInIdentifier } from '@logto/schemas';
+import { useCallback } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+
+import { useConfirmModal } from '@/hooks/use-confirm-modal';
+import useLinkSocial from '@/hooks/use-social-link-account';
+import type { VerificationCodeIdentifier } from '@/types';
+import { formatPhoneNumberWithCountryCallingCode } from '@/utils/country-code';
+
+const useLinkSocialConfirmModal = () => {
+  const { show } = useConfirmModal();
+  const { t } = useTranslation();
+  const linkWithSocial = useLinkSocial();
+  const navigate = useNavigate();
+
+  return useCallback(
+    async (method: VerificationCodeIdentifier, target: string, connectorId: string) => {
+      const [confirm] = await show({
+        confirmText: 'action.bind_and_continue',
+        cancelText: 'action.change',
+        cancelTextI18nProps: {
+          method: t(`description.${method === SignInIdentifier.Email ? 'email' : 'phone_number'}`),
+        },
+        ModalContent: t('description.link_account_id_exists', {
+          type: t(`description.${method === SignInIdentifier.Email ? 'email' : 'phone_number'}`),
+          value:
+            method === SignInIdentifier.Email
+              ? target
+              : formatPhoneNumberWithCountryCallingCode(target),
```

Comment:
It seems it would be safer if we reverse the logic, what do you think? Format the phone number only if the sign in method is phone

Reply:
update gloably